### PR TITLE
Feature/sort by distance

### LINKED ISSUE #383: Display event distance to user
Display the distance from events to the user in the HomeScreen's list mode.

### PR DESCRIPTION
**Pull request by alejandrocalles**
Sort events by distance.

Display distance to user in `EventListItem`, but only on the `HomeScreen`.

Closes #383 and #384.

### CODE REVIEW

File: app/src/main/java/com/github/swent/echo/compose/components/ListDrawer.kt [hunk 1]
```diff
@@ -124,7 +127,15 @@ fun EventListItem(
...
     val association = event.organizer?.name?.let { "$it • " } ?: ""
     // Format the distance to be displayed
-    val dist = distance?.let { "${it}km • " } ?: ""
+    val dist =
+        distance
+            ?.let {
+                val fmt = { format: String, value: Double ->
+                    String.format(Locale.getDefault(), format, value)
+                }
+                if (it < 1000) fmt("%.0fm", it) else fmt("%.1fkm", it / 1000.0)
```
Comment (srsingh04) [inline app/src/main/java/com/github/swent/echo/compose/components/ListDrawer.kt:136]: Knitpick: good practise to not Magic values in the code: 1000.0 can be extracted into a variable?

**Comment by alejandrocalles** [comment]
User should be able to sort event by distance ascending or descending. By default, events are sorted by date, so the distance sort is applied on top.

File: app/src/main/java/com/github/swent/echo/compose/components/ListDrawer.kt [hunk 2]
```diff
@@ -73,22 +75,23 @@ fun ListDrawer(
...
+                    selectedEvent = selectedEvent,
+                    isOnline = isOnline,
+                    refreshEvents = refreshEvents,
+                    viewOnMap = viewOnMap,
+                    modify = modify,
+                    onTagPressed = onTagPressed,
+                    canModifyEvent = canModifyEvent,
+                    distance = userLocation?.let { event.location.toLatLng().distanceTo(it) },
+                )
+                Spacer(modifier = Modifier.height(12.dp))
```
Comment (srsingh04) [inline app/src/main/java/com/github/swent/echo/compose/components/ListDrawer.kt:93]: Similarly for the height value, could be extracted into a variable to b e able to make modifications easily by someone else in the future

**Review by srsingh04 — APPROVED** [review]
LGTM! Left some minor comments, but can be merged into the main regardless due to time constraints.

**Comment by alejandrocalles** [comment]
Those are good points! I have addressed them.

**Review by unglazedstamp — APPROVED** [review]
The changes requested by @srsingh04 have been made

**Pull request merged**